tracer/tracer.mli

```diff
@@ -0,0 +1,30 @@
+type t
+
+type trace_layers =
+  { app : bool
+  ; lsp : bool
+  ; jsonrpc : bool
+  }
+
+val trace_layers :
+  ?app:unit -> ?lsp:unit -> ?jsonrpc:unit -> unit -> trace_layers
+
+val create : max_n_last_events:int -> trace_layers -> t
+
+val trace_lsp_recv_req :
+     ?name:string
+  -> t option
+  -> id:[ `String of string | `Int of int ]
```

Comment:
You can use `Jsonrpc.Id.t` directly here and elsewhere.

Reply:
I am afraid of creating a cycle between `tracer` and `jsonrpc`. I want to trace events in jrpc with the tracer, if that makes sense.

Reply:
`jsonrpc` has only type definitions, so I don't see how it could ever depend back on tracer.

Reply:
ah right! will fix where appropriate